### PR TITLE
api/v1: fix response format tests

### PR DESCRIPTION
**Pull request by fabxc**
@grobie @juliusv 

Tests existed but `httptest.ResponseRecorder` does not record the actual response, which I expected.

### CODE REVIEW

File: web/api/v1/api_test.go [hunk 1]
```diff
@@ -321,20 +322,31 @@ func TestEndpoints(t *testing.T) {
...
+	defer resp.Body.Close()
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("Error reading response body: %s", err)
 	}
 
-	if h := w.Header().Get("Content-Type"); h != "application/json" {
-		t.Fatalf("expected Content-Type %q but got %q", "application/json", h)
+	if resp.StatusCode != 200 {
+		t.Fatalf("Return code %d expected in success response but got %d", 422, resp.StatusCode)
```
Comment (juliusv) [inline web/api/v1/api_test.go]: s/422/200/?

File: web/api/v1/api_test.go [hunk 2]
```diff
@@ -347,20 +359,31 @@ func TestRespondSuccess(t *testing.T) {
...
+	defer resp.Body.Close()
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("Error reading response body: %s", err)
 	}
 
-	if h := w.Header().Get("Content-Type"); h != "application/json" {
-		t.Fatalf("expected Content-Type %q but got %q", "application/json", h)
+	if resp.StatusCode != 422 {
+		t.Fatalf("Return code %d expected in success response but got %d", 422, resp.StatusCode)
```
Comment (juliusv) [inline web/api/v1/api_test.go]: s/success/error/?

File: web/api/v1/api_test.go (hunk 1 above)
Comment (fabxc) [inline web/api/v1/api_test.go]: Done.

File: web/api/v1/api_test.go (hunk 2 above)
Comment (fabxc) [inline web/api/v1/api_test.go]: Done.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**